packages/adaptive_scaffold/lib/src/slot_layout.dart

```diff
@@ -0,0 +1,188 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+import 'breakpoints.dart';
+
+/// A Widget that takes a mapping of [SlotLayoutConfig]s to [Breakpoint]s and
+/// adds the appropriate Widget based on the current screen size.
+///
+/// See also:
+/// * [AdaptiveLayout], where [SlotLayout]s are assigned to placements on the
+/// screen called "slots".
+class SlotLayout extends StatefulWidget {
+  /// Creates a [SlotLayout] widget.
+  const SlotLayout({required this.config, super.key});
+
+  /// Given a context and a config, it returns the [SlotLayoutConfig] that will
+  /// be chosen from the config under the context's conditions.
+  static SlotLayoutConfig? pickWidget(
```

Comment:
It doesn't seem to me that this function needs to be public.

Reply:
The method is used within some logic in AdaptiveLayout.